### ISSUE TITLE
Reading an object with a single resource entry from Leshan gives error

### ISSUE THREAD

**Issue by pasear** [eclipse/wakaama#194]
I tried to start with the simplest [test_object](https://github.com/eclipse/wakaama/blob/master/examples/lightclient/test_object.c)

For simplicity, I commented out all methods except read. This does not affect the behavior of read.

```
        testObj->readFunc = prv_read;
/*
        testObj->writeFunc = prv_write;
        testObj->executeFunc = prv_exec;
        testObj->createFunc = prv_create;
        testObj->deleteFunc = prv_delete;
        testObj->discoverFunc = prv_discover;
*/
```

In `prv_read()`, I try to start with only one resource entry. So I changed the code to

```
if (*numDataP == 0)
    {
        *dataArrayP = lwm2m_data_new(1);
        *numDataP = 1;
        (*dataArrayP)[0].id = 1;
}
```

Then, I try to read the whole instance, say 20100/0, from a Leshan demo serveer.
The server shows 400 error upon this read.
(It seems that the path.getResourceId() is null)

But if I read 20100/0/1, it works as expected.
Defining two entries also works.

```
if (*numDataP == 0)
    {
        *dataArrayP = lwm2m_data_new(2);
        *numDataP = 2;
        (*dataArrayP)[0].id = 1;
        (*dataArrayP)[1].id = 3;
}
```
Could this possibly be a bug? Or did I do something wrong?
Thank you.

**Comment by dnav** [comment]
Hi,

After some testing, there is indeed an issue with single resource objects when the media type is not specified in the request.
I made a quick fix in #193 

Regards

**Comment by pasear** [comment]
Hi,

I cherry-picked the patch and it works! Thanks!

**Comment by dnav** [comment]
Fixed by #193
